Sources/EduIDExpansion/Controllers/Scanning/ScanViewController.swift

```diff
@@ -122,12 +103,23 @@ Scan it here
         upperDarkView.size(to: lowerDarkView)
         middelLeftDarkView.size(to: middelRightDarkView)
     }
-}
-
-//MARK: - preview layer delegate
-extension ScanViewController: AVCaptureMetadataOutputObjectsDelegate {
     
-    func metadataOutput(_ output: AVCaptureMetadataOutput, didOutput metadataObjects: [AVMetadataObject], from connection: AVCaptureConnection) {
-        print(metadataObjects)
+    //MARK: - toggle flash action
+    @objc
+    func toggleTorch() {
+        guard let device = AVCaptureDevice.default(for: AVMediaType.video) else { return }
+        guard device.hasTorch else { print("Torch isn't available"); return }
+
+        do {
+            try device.lockForConfiguration()
+            let on = flashButton.image(for: .normal) == .flashLightOff
```

Comment:
Storing state in a control isn't very clean.

Reply:
I moved the state into variable flashIsOn: Bool